### PR TITLE
CI: fix release builds

### LINKED ISSUE #886: Cannot download 1.8.0 with modules
Well, I just tried to update my project's dependencies, and things broke apart.
Seems like its not possible to get grpc-gateway at the moment:

### Steps you follow to reproduce the error:

```bash
$ go get -u github.com/grpc-ecosystem/grpc-gateway
go: github.com/go-resty/resty@v1.9.0: parsing go.mod: unexpected module path "gopkg.in/resty.v1"
go: finding golang.org/x/sys latest
go: finding golang.org/x/net latest
go get: error loading module requirements
```

FYI, `go get github.com/grpc-ecosystem/grpc-gateway@v1.8.0` is just as broken.

### What did you expect to happen instead:

Work?

### What's your theory on why it isn't working:

grpc-gateway is using the wrong import path - its seems the [replace directive](https://github.com/grpc-ecosystem/grpc-gateway/blob/master/go.mod#L18) in go.mod does not propagate properly?

https://github.com/go-resty/resty#installation
https://github.com/go-resty/resty/issues/230

Not directly related issue, but a discussion and reference for impact of "unexpected module path":
https://github.com/golang/lint/issues/436

**EDIT:** I was a bit fast on naming the issue - please correct as seen fit.

### PR DESCRIPTION
**Pull request by johanbrandhorst**
For some reason running "go mod vendor" edited the sum file.
Lets make sure there are no more surprises by running it in
CI as well.

Fixes #886

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -95,6 +96,7 @@ workflows:
       - release:
           filters:
             branches:
-              ignore: /.*/
-            tags:
-              only: /v[0-9]+(\.[0-9]+)*(-.*)*/
+              #ignore: /.*/
+              only: fix-release-builds
+            #tags:
+            #only: /v[0-9]+(\.[0-9]+)*(-.*)*/
```
Comment (johanbrandhorst) [inline .circleci/config.yml]: REVERT ME AFTER TESTING

**Comment by johanbrandhorst** [comment]
I rebuilt the build environment image with go 1.12 which explains some of the other errors.

**Comment by johanbrandhorst** [comment]
Gonna leave the updating of the build environment for another PR.

**Comment by johanbrandhorst** [comment]
> This looks good from _very_ quick look.
> 
> One thing you could do to help test it before going 'live' is to first tag it with a semver prerelease tag like `v1.8.1-pre1`, which means someone can use that as part of a `go get github.com/grpc-ecosystem/grpc-gateway@v1.8.1-pre1`, but it won't yet be live for someone that does not use the prelease tag.

Excellent idea, I shall action that as soon as I've merged this.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -29,7 +29,7 @@ docker run -v $(pwd):/src/grpc-gateway --rm jfbrandhorst/grpc-gateway-build-env
         make realclean && \
         make examples SWAGGER_CODEGEN="${SWAGGER_CODEGEN}"'
 docker run -itv $(pwd):/grpc-gateway -w /grpc-gateway --entrypoint /bin/bash --rm \
-    l.gcr.io/google/bazel -c 'bazel run :gazelle_fix; bazel run :buildifier'
+    l.gcr.io/google/bazel -c 'bazel run :gazelle fix; bazel run :buildifier'
```
Comment (achew22) [inline CONTRIBUTING.md]: I believe this should just be `bazel run :gazelle`

Comment (johanbrandhorst) [reply CONTRIBUTING.md]: Ah ok, I'll update the CI configuration too

**Pull request merged**